Lib/test/test_os.py

```diff
@@ -87,6 +88,43 @@ def test_getcwd(self):
         cwd = os.getcwd()
         self.assertIsInstance(cwd, str)
 
+    def test_getcwd_long_path(self):
+        # bpo-37412: On Linux, PATH_MAX is usually around 4096 bytes. On
+        # Windows, MAX_PATH is defined as 260 characters, but the universal
+        # naming convention (UNC) allows longer paths. Internally, the os
+        # module uses MAXPATHLEN which is at least 1024.
+        min_len = 2000   # characters
+        dirlen = 200     # characters
+        dirname = 'python_test_dir_'
+        dirname = dirname + ('a' * (dirlen - len(dirname)))
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            with support.change_cwd(tmpdir):
+                path = tmpdir
+                expected = path
+
+                while True:
+                    cwd = os.getcwd()
+                    self.assertEqual(cwd, expected)
+
+                    need = min_len - (len(cwd) + len(os.path.sep))
+                    if need <= 0:
+                        break
+                    if len(dirname) > need and need > 0:
+                        dirname = dirname[:need]
+
+                    path = os.path.join(path, dirname)
+                    try:
+                        os.mkdir(path)
+                    except OSError as exc:
+                        if exc.errno == errno.ENAMETOOLONG:
```

Comment:
The CRT in Windows doesn't use `ENAMETOOLONG`. In the Windows API, depending on the context, a path that's too long fails with either `ERROR_PATH_NOT_FOUND` (3) or `ERROR_FILENAME_EXCED_RANGE` (206). Both  get mapped to the errno value `ENOENT`.

Reply:
I modified my PR to catch also FileNotFoundError. The test stops on such error: we reached the maximum path length.

Reply:
> The CRT in Windows doesn't use ENAMETOOLONG.

"if exc.errno == errno.ENAMETOOLONG:" is for Linux ;-)

Reply:
> > The CRT in Windows doesn't use ENAMETOOLONG.
> 
> "if exc.errno == errno.ENAMETOOLONG:" is for Linux ;-)

I know. My comment was written before the code handled `FileNotFoundError` (from `ENOENT`), so I assumed you thought the Windows CRT uses `ENAMETOOLONG`. It defines this error code, but never uses it.